internal/msalbase/account_test.go

```diff
@@ -79,12 +93,18 @@ func TestAccountConvertToJSONMap(t *testing.T) {
 		"username":         "user",
 		"extra":            "extra",
 	}
-	actualJSONMap, err := acc.ConvertToJSONMap()
+	b, err := json.Marshal(acc)
 	if err != nil {
-		t.Errorf("Error should be nil, but it is %v", err)
+		panic(err)
 	}
-	if !reflect.DeepEqual(jsonMap, actualJSONMap) {
-		t.Errorf("JSON account %+v differs from expected JSON account %+v", jsonMap, actualJSONMap)
+
+	got := map[string]interface{}{}
+	if err := stdJSON.Unmarshal(b, &got); err != nil {
+		panic(err)
```

Comment:
any reason these are preferred to be panics over asserts? Assuming because it's core functionality and nothing really with the MSAL functionality itself?

Reply:
I'm going to assume by assert you mean t.Fatalf().  So this is something that is more my style than a Go thing really.  

But it is as exactly as you say, setup code I panic in because that code is not part of the test.  At a quick glance it tells me the test is completely broken vs an unexpected outcome from the test.

Reply:
Makes sense, in MSAL.NET we used to do `Assert.Inconclusive("Test setup error ...")` but reverted to `Assert.Fail("Test setup error")` because the Inconclusive assert was too obscure. 

As long as the CI system interprets this as a failure and a developer looking at the test failures / logs easily understand what the problem is, it's good to go.